File: app/src/main/java/id/rizkimandiri/jadwalpelajaran/MainActivity.java
package id.rizkimandiri.jadwalpelajaran;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.webkit.CookieManager;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.MobileAds;

public class MainActivity extends AppCompatActivity {

  private Context contextPop;
  private WebView webViewPop;
  private AlertDialog builder;

  private String url = "file:///android_asset/index.html";

  private WebView webView;
  private String userAgent;

  private AdView adView;

  @Override
  protected void onCreate(Bundle savedInstanceState) {

      super.onCreate(savedInstanceState);
      setContentView(R.layout.activity_main);

      ///ca-app-pub-3940256099942544/6300978111 test
      MobileAds.initialize(this, "ca-app-pub-8641844379696896/4515918796");

      adView = findViewById(R.id.adView);
      AdRequest adRequest = new AdRequest.Builder().build();
      adView.loadAd(adRequest);

      webView = findViewById(R.id.webView);
      webView.setWebViewClient(new WebViewClient());
      webView.loadUrl(url);

      WebSettings webSettings = webView.getSettings();
      webSettings.setJavaScriptEnabled(true);

      // Set User Agent
      userAgent = System.getProperty("http.agent");
      webSettings.setUserAgentString(userAgent + "WebViewApp Jadwal Pelajaran/1");

      // Enable Cookies
      CookieManager.getInstance().setAcceptCookie(true);
      if(android.os.Build.VERSION.SDK_INT >= 21)
        CookieManager.getInstance().setAcceptThirdPartyCookies(webView, true);

      // WebView Tweaks
      webSettings.setRenderPriority(WebSettings.RenderPriority.HIGH);
      webSettings.setCacheMode(WebSettings.LOAD_DEFAULT);
      webSettings.setAppCacheEnabled(true);
      webSettings.setDomStorageEnabled(true);
      webSettings.setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NARROW_COLUMNS);
      webSettings.setUseWideViewPort(true);
      webSettings.setSaveFormData(true);
      webSettings.setEnableSmoothTransition(true);
      webView.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY);

      // Handle Popups
      webView.setWebChromeClient(new CustomChromeClient());
      webSettings.setJavaScriptCanOpenWindowsAutomatically(true);
      webSettings.setSupportMultipleWindows(true);
      contextPop = this.getApplicationContext();
  }

  @Override
  public void onBackPressed() {

    if(webView.canGoBack()) {
      webView.goBack();
    }
    else {
      //super.onBackPressed();
      // Terminate the app
      finishAffinity();
      System.exit(0);
    }
  }

  class CustomChromeClient extends WebChromeClient {

    @Override
    public boolean onCreateWindow(WebView view, boolean isDialog,
                                  boolean isUserGesture, Message resultMsg) {
      webViewPop = new WebView(contextPop);
      webViewPop.setVerticalScrollBarEnabled(false);
      webViewPop.setHorizontalScrollBarEnabled(false);
      webViewPop.setWebChromeClient(new CustomChromeClient());
      webViewPop.getSettings().setJavaScriptEnabled(true);
      webViewPop.getSettings().setSaveFormData(true);
      webViewPop.getSettings().setEnableSmoothTransition(true);
      webViewPop.getSettings().setUserAgentString(userAgent + "WebViewApp Jadwal Pelajaran/1");

      // set the WebView as the AlertDialog.Builder’s view
      builder = new AlertDialog.Builder(MainActivity.this, AlertDialog.THEME_DEVICE_DEFAULT_LIGHT).create();
      builder.setTitle("");
      builder.setView(webViewPop);

      builder.setButton("Close", new DialogInterface.OnClickListener() {
        @Override
        public void onClick(DialogInterface dialog, int id) {
          webViewPop.destroy();
          dialog.dismiss();
        }
      });

      builder.show();
      builder.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE|WindowManager.LayoutParams.FLAG_ALT_FOCUSABLE_IM);

      CookieManager cookieManager = CookieManager.getInstance();
      cookieManager.setAcceptCookie(true);
      if(android.os.Build.VERSION.SDK_INT >= 21) {
        cookieManager.setAcceptThirdPartyCookies(webViewPop, true);
        cookieManager.setAcceptThirdPartyCookies(webView, true);
      }

      WebView.WebViewTransport transport = (WebView.WebViewTransport) resultMsg.obj;
      transport.setWebView(webViewPop);
      resultMsg.sendToTarget();

      return true;
    }

    @Override
    public void onCloseWindow(WebView window) {
      //Toast.makeText(contextPop,"onCloseWindow called",Toast.LENGTH_SHORT).show();
      try {
        webViewPop.destroy();
      } catch (Exception e) {
        Log.d("Webview Destroy Error: ", e.getStackTrace().toString());
      }

      try {
        builder.dismiss();
      } catch (Exception e) {
        Log.d("Builder Dismiss Error: ", e.getStackTrace().toString());
      }

    }
  }
}

File: app/src/main/java/id/rizkimandiri/jadwalpelajaran/MainActivity.kt
package id.rizkimandiri.jadwalpelajaran

import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.KeyEvent
import android.view.MotionEvent
import android.view.View
//import android.webkit.WebSettings.*
import android.webkit.WebView
import android.webkit.WebViewClient
//import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.ads.*

class MainActivity : AppCompatActivity() {

    lateinit var adView : AdView

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /* AdMob Action */
        // Set Test Device
        val testDeviceIds = listOf("114D091401CBB4D34FB80E68FE66F1ED")
        val configuration = RequestConfiguration.Builder().setTestDeviceIds(testDeviceIds).build()
        MobileAds.setRequestConfiguration(configuration)
        // Load an ad into the AdMob banner view.
        adView = findViewById<View>(R.id.adView) as AdView
        val adRequest = AdRequest.Builder().build()
        adView.loadAd(adRequest)

        /* AdMob Listener
        adView.adListener = object : AdListener(){
            override fun onAdFailedToLoad(p0: LoadAdError) {
                super.onAdFailedToLoad(p0)
                val toastMessage = "ad fail to load"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }
            override fun onAdLoaded() {
                super.onAdLoaded()
                val toastMessage = "ad loaded"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }
            override fun onAdOpened() {
                super.onAdOpened()
                val toastMessage = "ad is open"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }
            override fun onAdClicked() {
                super.onAdClicked()
                val toastMessage = "ad is clicked"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }

            override fun onAdClosed() {
                super.onAdClosed()
                val toastMessage = "ad is closed"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }
            override fun onAdImpression() {
                super.onAdImpression()
                val toastMessage = "ad impression"
                Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_LONG).show()
            }
        }*/
        /* AdMob Action */

        val mWebView = findViewById<View>(R.id.WebView) as WebView
        mWebView.loadUrl("file:///android_asset/index.html")

        val webSetting = mWebView.settings
        webSetting.javaScriptEnabled = true
        webSetting.allowFileAccess = true
        webSetting.domStorageEnabled = true
        webSetting.allowFileAccess = true

        mWebView.webViewClient = WebViewClient()
        mWebView.scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY
        mWebView.isVerticalScrollBarEnabled = false
        mWebView.isHorizontalScrollBarEnabled = false

        class MyWebViewClient : WebViewClient() {

            override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
                if (url != null && (url.startsWith("file:///"))) {
                    return false
                }else{
                    // Otherwise, the link is not for a page on my site, so launch another Activity that handles URLs
                    Intent(Intent.ACTION_VIEW, Uri.parse(url)).apply {
                        startActivity(this); return true
                    }
                }
            }
        }
        val myWebView: WebView = mWebView
        myWebView.webViewClient = MyWebViewClient()


        mWebView.canGoBack()
        mWebView.setOnKeyListener(View.OnKeyListener{ _, keyCode, event ->
            if (keyCode == KeyEvent.KEYCODE_BACK

                && event.action == MotionEvent.ACTION_UP
                && mWebView.canGoBack()){
                mWebView.goBack()
                return@OnKeyListener true
            }
            false
        })
    }

    override fun onDestroy() {
        adView.destroy()
        super.onDestroy()
    }
}